docs/scalars_and_keras.ipynb

```diff
@@ -440,28 +425,374 @@
+      },
+      "outputs": [],
+      "source": [
+        "log_dir = 'logs/batch_level/' + datetime.now().strftime(\"%Y%m%d-%H%M%S\") + '/train'\n",
+        "train_writer = tf.summary.create_file_writer(log_dir)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "YlVZY2VqrK9D"
+      },
+      "source": [
+        "To enable batch-level logging, custom `tf.summary` metrics should be defined by overriding `train_step()` in the Model's class definition and enclosed in a summary writer context. This can simply be made combined into subclassed Model definitions or can extend to edit our previous Functional API Model, as shown below:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "IGcNr1ZS1xXL"
+      },
+      "outputs": [],
+      "source": [
+        "class MyModel(tf.keras.Model):\n",
+        "  def __init__(self, model):\n",
+        "    super().__init__()\n",
+        "    self.model = model\n",
+        "  \n",
+        "  def train_step(self, data):\n",
+        "    x, y = data\n",
+        "    with tf.GradientTape() as tape:\n",
+        "      y_pred = self.model(x, training=True)\n",
+        "      loss = self.compiled_loss(y, y_pred)\n",
+        "      mse = tf.keras.losses.mean_squared_error(y, K.max(y_pred, axis=-1))\n",
+        "    self.optimizer.minimize(loss, self.trainable_variables, tape=tape)\n",
+        "    with train_writer.as_default(step=self._train_counter):\n",
+        "      tf.summary.scalar('batch_loss', loss)\n",
+        "      tf.summary.scalar('batch_mse', mse)\n",
+        "    return self.compute_metrics(x, y, y_pred, None)\n",
+        "  \n",
+        "  def call(self, x):\n",
+        "    x = self.model(x)\n",
+        "    return x\n",
+        "\n",
+        "# Adds custom batch-level metrics to our previous Functional API model\n",
+        "model = MyModel(create_model())\n",
+        "model.compile(optimizer='adam',\n",
+        "              loss='sparse_categorical_crossentropy',\n",
+        "              metrics=['accuracy'])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IlqPZ-h_1_0z"
+      },
+      "source": [
+        "Setting up a summary writer to a different log directory:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Zjv_7dRwiyf9"
+      },
+      "outputs": [],
+      "source": [
+        "log_dir = 'logs/batch_level/' + datetime.now().strftime(\"%Y%m%d-%H%M%S\") + '/train'\n",
+        "train_writer = tf.summary.create_file_writer(log_dir)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "8wqqNaXP2bLc"
+      },
+      "source": [
+        "Define our Tensorboard callback to log both epoch-level and batch-level metrics to our log directory and call `model.fit()` with our selected `batch_size`: "
```

Comment:
nit: can you capitalize as "TensorBoard" (capital B) here and in the other markdown cells below?

Reply:
Fixed.